app/services/duplicate_application.rb

```diff
@@ -45,6 +45,11 @@ def duplicate
       )
     end
 
+    number_of_references_has_changed = original_application_form.application_references.size != new_application_form.application_references.size
+    if number_of_references_has_changed
+      new_application_form.update! references_completed: false
```

Comment:
If the original had say 1 declined reference and 2 approved references then we'd strip out the declined one leaving the 2 approved ones. So even if the size changes I think you could end up in a 'completed' state. Could we check for the presence of at least 2 references in the new application form?

Reply:
That makes sense!

Reply:
Hadn't thought of using `can_add_reference?` :+1: